starlarkstruct/struct.go

```diff
@@ -268,10 +237,13 @@ func structsEqual(x, y *Struct, depth int) (bool, error) {
 		return false, nil
 	}
 
-	for i, n := 0, x.len(); i < n; i++ {
-		if x.entries[i].name != y.entries[i].name {
+	for k, xv := range x.members {
```

Comment:
This loop is now nondeterministic because EqualDepth may fail before or after an unequal field comparison, depending on iteration order. I don't see any way to make it deterministic other than by iterating over Keys(), which could increase the cost of comparison quite a lot, since it allocates.

Perhaps you could add benchmarks of these operations to testdata/benchmark.star?